### PR TITLE
Refactor instance segmentation and tiling inference

### PR DESCRIPTION
**Pull request by sovrasov**
### Summary

IS OV wrapper and tilers are imported from ModelAPI.

### Checklist

- [ ] I have added unit tests to cover my changes.​
- [ ] I have added integration tests to cover my changes.​
- [ ] I have added e2e tests for validation.
- [ ] I have added the description of my changes into CHANGELOG in my target branch (e.g., [CHANGELOG](https://github.com/openvinotoolkit/training_extensions/blob/develop/CHANGELOG.md) in develop).​
- [ ] I have updated the documentation in my target branch accordingly (e.g., [documentation](https://github.com/openvinotoolkit/training_extensions/tree/develop/docs) in develop).
- [ ] I have [linked related issues](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword).

### License

- [ ] I submit _my code changes_ under the same [Apache License](https://github.com/openvinotoolkit/training_extensions/blob/develop/LICENSE) that covers the project.
      Feel free to contact the maintainers if that's a concern.
- [ ] I have updated the license header for each file (see an example below).

```python
# Copyright (C) 2023 Intel Corporation
# SPDX-License-Identifier: Apache-2.0
```

### CODE REVIEW

**Review by eunwoosh — DISMISSED** [review]
LGTM. I have a question. `OTXMaskRCNNModel` seems not to be used now. Why isn't it removed?

File: src/otx/api/usecases/exportable_code/prediction_to_annotation_converter.py [hunk 1]
```diff
@@ -477,19 +485,21 @@ def convert_to_annotation(self, predictions: tuple, metadata: Dict[str, Any]) ->
...
             if self.use_ellipse_shapes:
-                shape = convert_bbox_to_ellipse(box[0] / width, box[1] / height, box[2] / width, box[3] / height)
+                shape = convert_bbox_to_ellipse(
+                    obj.xmin / width, obj.ymin / height, obj.xmax / width, obj.ymax / height
+                )
                 annotations.append(
                     Annotation(
                         shape,
-                        labels=[ScoredLabel(self.labels[int(class_idx) - 1], float(score))],
+                        labels=[ScoredLabel(self.labels[int(obj.id) - 1], float(obj.score))],
```
Comment (goodsong81) [inline src/otx/api/usecases/exportable_code/prediction_to_annotation_converter.py:498]: Does this `id` mean `class_id`? Might be misleading.

**Review by goodsong81 — COMMENTED** [review]
LGTM but I'd bring @eugene123tw in.

File: src/otx/api/usecases/exportable_code/prediction_to_annotation_converter.py (hunk 1 above)
Comment (sovrasov) [reply src/otx/api/usecases/exportable_code/prediction_to_annotation_converter.py:498]: Yes, in ModelAPI result objects id = class_id

**Comment by sovrasov** [comment]
> LGTM. I have a question. `OTXMaskRCNNModel` seems not to be used now. Why isn't it removed?

This was a merge artifact, now it's finally removed.

File: src/otx/api/usecases/exportable_code/prediction_to_annotation_converter.py (hunk 1 above)
Comment (goodsong81) [reply src/otx/api/usecases/exportable_code/prediction_to_annotation_converter.py:498]: Hope it could be fixed someday in ModelAPI repo... :)

**Review by eugene123tw — APPROVED** [review]
LGTM. Thanks for adding tiler modules to model API!

**Review by eunwoosh — APPROVED** [review]

**Review by jaegukhyun — APPROVED** [review]

**Pull request merged**